src/Hl7.Fhir.Patch/Internal/IAdapter.cs

```diff
@@ -0,0 +1,50 @@
+﻿/* 
+ * Copyright (c) 2020, Firely (info@fire.ly) and contributors
+ * See the file CONTRIBUTORS for details.
+ * 
+ * This file is licensed under the BSD 3-Clause license
+ * available at https://github.com/FirelyTeam/fhir-net-api/blob/master/LICENSE
+ */
+
+using Hl7.Fhir.Specification;
+
+namespace Hl7.Fhir.Patch.Internal
+{
+    /// <summary>
+    /// This API supports infrastructure and is not intended to be used
+    /// directly from your code. This API may change or be removed in future releases.
+    /// </summary>
+    public interface IAdapter
```

Comment:
Naming: `Adapter` suggests the use of the Adapter pattern, which is not the case here. Something with Editor, Updater or maybe even Patcher in it might work better.